storage/rocksdb/ha_rocksdb.cc

```diff
@@ -15501,6 +15518,9 @@ int ha_rocksdb::mrr_fill_buffer() {
     return HA_ERR_INTERNAL_ERROR;
   }
 
+  THD *thd = table->in_use;
+  ssize_t elements_limit = THDVAR(thd, mrr_batch_size);
+  n_elements = std::min(n_elements, elements_limit);
```

Comment:
This will cause a buffer overrun when one sets @@rocksdb_mrr_batch_size=0 (which the patch allows)
There's no meaningful reason to use this setting but this should not result in a buffer overrun anyway...

I think these lines should be moved to be above the `if (n_elements) < 1` above.

I think we should either adjust the variable limit, or have multi_range_read_info() switch to native MRR in this case.